pkg/microservice/systemconfig/core/features/service/feature.go

```diff
@@ -0,0 +1,102 @@
+package service
+
+import (
+	"context"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/koderover/zadig/pkg/microservice/systemconfig/core/features/repository/models"
+	"github.com/koderover/zadig/pkg/microservice/systemconfig/core/features/repository/mongodb"
+)
+
+type Feature string
+
+const (
+	ModernWorkflow             Feature = "ModernWorkflow"
+	CommunityProjectRepository Feature = "CommunityProjectRepository"
+	RegisterTrigger            Feature = "RegisterTrigger"
+)
+
+var Features FeatureGates = map[Feature]bool{
+	ModernWorkflow:             false,
+	CommunityProjectRepository: false,
+	RegisterTrigger:            true,
+}
+
+type FeatureGates map[Feature]bool
+
+func (fg FeatureGates) EnabledFeatures() []Feature {
+	var res []Feature
+
+	for k, v := range fg {
+		if v {
+			res = append(res, k)
+		}
+	}
+
+	return res
+}
+
+func (fg FeatureGates) FeatureEnabled(f Feature) bool {
+	return fg[f]
+}
+
+func (fg FeatureGates) MergeFeatureGates(fs ...FeatureGates) {
```

Comment:
add comment to notice that latter feature configs will be overrided by former ones